beamer/tests/contracts/test_fee_sub.py

```diff
@@ -0,0 +1,155 @@
+import ape
+from hexbytes import HexBytes
+from web3.constants import ADDRESS_ZERO
+
+from beamer.tests.agent.utils import make_address
+from beamer.tests.util import alloc_accounts, alloc_whitelisted_accounts, make_request
+
+
+def test_fee_sub(fee_sub, token, request_manager):
+    mint_amount = 5
+    sent_amount = 2
+    chain_id = ape.chain.chain_id
+    (sender, target) = alloc_accounts(2)
+
+    wrong_token_address = make_address()
+
+    with ape.reverts("Token not to be subsidized"):
+        fee_sub.createRequest(
+            chain_id,
+            wrong_token_address,
+            wrong_token_address,
+            target,
+            sent_amount,
+            1800,
+            sender=sender,
+        )
+
+    with ape.reverts("Transfer amount too small to be subsidized"):
+        fee_sub.createRequest(
+            chain_id, token.address, token.address, target, 1, 1800, sender=sender
+        )
+
+    token.mint(sender, mint_amount, sender=sender)
+
+    with ape.reverts("Insufficient allowance"):
+        fee_sub.createRequest(
+            chain_id, token.address, token.address, target, 2, 1800, sender=sender
+        )
+
+    token.approve(fee_sub.address, mint_amount, sender=sender)
+
+    assert token.balanceOf(sender) == mint_amount
+
+    request = fee_sub.createRequest(
+        chain_id, token.address, token.address, target, sent_amount, 1800, sender=sender
+    )
+
+    request_id = request.return_value
+
+    assert token.balanceOf(sender) == mint_amount - sent_amount  # no fee
```

Comment:
can you also assert that the `fee_sub_balance == old_balance - sent_amount - total_fee(sent_amount)`

Reply:
is this really true? old balance - total_fee is ok to me and for the sent_amount we are already getting it and sending it so no storing in there

Reply:
ah you are right